packages/grafana-data/src/field/fieldColor.ts

```diff
@@ -234,3 +235,26 @@ function getFixedColor(field: Field, theme: GrafanaTheme2) {
     return theme.visualization.getColorByName(field.config.color?.fixedColor ?? FALLBACK_COLOR);
   };
 }
+
+function getUniqueColorsByValue(): FieldColorMode {
+  return {
+    id: 'unique-colors-by-value',
+    name: 'Unique colors by value',
+    description: 'Assigns each value a unique color',
+
+    getCalculator: (_field, theme) => {
+      let assigned: Map<any, number> = new Map<number, number>();
+      let next = 0;
+      let colors = theme.visualization.palette.map(theme.visualization.getColorByName);
+
+      return (value, percent, threshold) => {
+        if (!assigned.has(value)) {
+          assigned.set(value, next);
+          next = (next + 1) % colors.length;
+        }
+
+        return colors[assigned.get(value) || 0];
```

Comment:
How many colors are there to choose from? 

I can assume this will be very useful in state timeline which in most cases will not have that many unique states to color. I'm just wondering, if we have a finite set of colors and we don'e end up in a situation where different values will get the same color.

Reply:
It seems that there are [55 different colors in the classic palette](http://github.com/grafana/grafana/blob/7f882eea05d7054407cf2c1fa05cbfe38414aa05/packages/grafana-data/src/themes/createVisualizationColors.ts#L234-L289).

IMO, it's not a big deal if the colors start repeating after a while as it hasn't been a big deal until now when we had many series and they started to share colors.

If color repeating becomes an issue, we can iterate building a color generator that builds colors in the midpoints of the ones from this palette, but anyway we'll still reach a point where colors would be undistinguishable for the human eye.